client/actions/events/notifications.js

```diff
@@ -36,12 +36,37 @@ const onEventUnlocked = (_e, data) => (
             const itemLock = lockUtils.getLock(eventInStore, locks);
             const sessionId = selectors.general.session(getState()).sessionId;
 
+            eventInStore = {
+                ...eventInStore,
+                _id: data.item,
+                lock_action: null,
+                lock_user: null,
+                lock_session: null,
+                lock_time: null,
+                _etag: data.etag,
+            };
+
+            dispatch({
+                type: EVENTS.ACTIONS.UNLOCK_EVENT,
+                payload: {event: eventInStore},
+            });
+
             // If this is the event item currently being edited, show popup notification
             if (itemLock !== null &&
                 data.lock_session !== sessionId &&
                 itemLock.session === sessionId
             ) {
                 const user = selectors.general.users(getState()).find((u) => u._id === data.user);
+                const autoSaves = selectors.forms.autosaves(getState());
+                let autoSaveInStore = get(autoSaves.event, data.item);
```

Comment:
These code changes (and reloading editor) are common for planning and event unlock. Could probably be extracted to a single action in main.js ?